public/src/components/dashboard/SubmissionsTable.tsx

```diff
@@ -27,66 +26,75 @@ const SubmissionsTable = () => {
+        if (state.enrollmentsByCourseID[courseID.toString()]?.status !== Enrollment_UserStatus.STUDENT) {
+            return
+        }
+        // Submissions are indexed by the assignment order - 1.
+        const submission = submissions.find(sub => sub.AssignmentID === assignment.ID) ?? create(SubmissionSchema)
+        const status = getStatusByUser(submission, state.self.ID)
+        if (!isApproved(status) && assignment.deadline) {
+            const date = timestampDate(assignment.deadline)
+            const now = new Date()
+
+            /*
+                Only show assignments which are the same year as now and if one month after deadline month.
+                This way students will only see assignments which are the current year and the expired assignments
+                are shown for a month after the deadline.
+            */
+            if (date.getFullYear() !== now.getFullYear() || date.getMonth() > now.getMonth() + 1) {
                 return
             }
-            // Submissions are indexed by the assignment order - 1.
-            const submission = submissions.find(sub => sub.AssignmentID === assignment.ID) ?? create(SubmissionSchema)
-            const status = getStatusByUser(submission, state.self.ID)
-            if (!isApproved(status) && assignment.deadline) {
-                const deadline = timeFormatter(assignment.deadline)
-                if (deadline.daysUntil > 3 && submission.score >= assignment.scoreLimit) {
-                    deadline.className = "table-success"
-                }
-                if (!deadline.message) {
-                    return
-                }
-                const course = state.courses.find(c => c.ID === courseID)
-                table.push(
-                    <tr key={assignment.ID.toString()} className={`clickable-row ${deadline.className}`}
-                        onClick={() => history.push(`/course/${courseID}/lab/${assignment.ID}`)}>
-                        <th scope="row">{course?.code}</th>
-                        <td>
-                            {assignment.name}
-                            {assignment.isGroupLab ?
-                                <span className="badge ml-2 float-right"><i className={Icon.GROUP} title="Group Assignment" /></span> : null}
-                        </td>
-                        <td><ProgressBar courseID={courseID.toString()} submission={submission} type={Progress.OVERVIEW} /></td>
-                        <td>{getFormattedTime(assignment.deadline, true)}</td>
-                        <td>{deadline.message ? deadline.message : '--'}</td>
-                        <td className={SubmissionStatus[status]}>
-                            {assignmentStatusText(assignment, submission, status)}
-                        </td>
-                    </tr>
-                )
-            }
-        })
-        return table
-    }
+
+            const deadline = timeFormatter(date, submission.score >= assignment.scoreLimit)
```

Comment:
Seems like this should be called `deadlineFormatter` instead, since it returns a `Deadline` interface.

Reply:
Probably should also revert the `date` input to `assignment.deadline` since `isExpired(assignment)` hides the `date` variable.

Reply:
Or you could pass the `assignment` and do the `submission.score >= assignment.scoreLimit` comparison internally in `deadlineFormatter`:
```tsx
const deadline = timeFormatter(assignment, submission.score)
```